test/e2e/suite/command/sign.go

```diff
@@ -0,0 +1,29 @@
+package command
+
+import (
+	. "github.com/notaryproject/notation/test/e2e/internal/notation"
+	"github.com/notaryproject/notation/test/e2e/internal/utils"
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var _ = Describe("notation sign", func() {
+	It("with JWS signature format", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.ReferenceWithTag(), "--signature-format", "jws").
+				MatchKeyWords("Successfully signed")
+
+			OldNotation().Exec("verify", artifact.ReferenceWithTag()).
+				MatchKeyWords("Successfully verified")
+		})
+	})
+
+	It("with COSE signature format", func() {
+		Host(BaseOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.ReferenceWithTag(), "--signature-format", "cose").
+				MatchKeyWords("Successfully signed")
+
+			OldNotation().Exec("verify", artifact.ReferenceWithTag()).
+				MatchKeyWords("Successfully verified")
+		})
+	})
```

Comment:
nit: can we write parameterized test which take jws and cose as argument?

Reply:
Good suggestion. It works well when we have many test cases with the same arguments. However, for siging test cases, we only have a few test cases with the same flags and the same key word matching. We can reduce the code length a little bit by using parameterized test, but we will lose readability a lot and hard to maintain or modify the test cases.

I prefer to keep it simple now. If we have too much test cases with the same arguments, I will update to use parameterized test later.